openpype/hosts/flame/plugins/publish/precollect_instances.py

```diff
@@ -0,0 +1,260 @@
+import pyblish
+import openpype
+import openpype.hosts.flame.api as opfapi
+from openpype.hosts.flame.otio import flame_export
+
+# # developer reload modules
+from pprint import pformat
+
+
+class PrecollectInstances(pyblish.api.ContextPlugin):
+    """Collect all Track items selection."""
+
+    order = pyblish.api.CollectorOrder - 0.49
+    label = "Precollect Instances"
+    hosts = ["flame"]
+
+    audio_track_items = []
+
+    def process(self, context):
+        project = context.data["flameProject"]
+        sequence = context.data["flameSequence"]
+        self.otio_timeline = context.data["otioTimeline"]
+        self.clips_in_reels = opfapi.get_clips_in_reels(project)
+        self.fps = context.data["fps"]
+
+        # process all sellected
+        with opfapi.maintained_segment_selection(sequence) as segments:
+            for segment in segments:
+                clip_data = opfapi.get_segment_attributes(segment)
+                clip_name = clip_data["segment_name"]
+                self.log.debug("clip_name: {}".format(clip_name))
+
+                # get openpype tag data
+                marker_data = opfapi.get_segment_data_marker(segment)
+                self.log.debug("__ marker_data: {}".format(
+                    pformat(marker_data)))
+
+                if not marker_data:
+                    continue
+
+                if marker_data.get("id") != "pyblish.avalon.instance":
+                    continue
+
+                # get file path
+                file_path = clip_data["fpath"]
+
+                # get source clip
+                source_clip = self._get_reel_clip(file_path)
+
+                first_frame = opfapi.get_frame_from_path(file_path) or 0
+
+                head, tail = self._get_head_tail(clip_data, first_frame)
+
+                # solve handles length
+                marker_data["handleStart"] = min(
+                    marker_data["handleStart"], head)
+                marker_data["handleEnd"] = min(
+                    marker_data["handleEnd"], tail)
+
+                # add audio to families
+                with_audio = False
+                if marker_data.pop("audio"):
+                    with_audio = True
+
+                # add tag data to instance data
+                data = {
+                    k: v for k, v in marker_data.items()
+                    if k not in ("id", "applieswhole", "label")
```

Comment:
Keys `("id", "applieswhole", "label")` are flame specific?

Reply:
good catch.. it had been inherited mess..

Reply:
Maybe just add comment so anybody reading the code knows why these are skipped.